### PR TITLE
fix(lint) make sure eslint also runs on TypeScript files

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Review by saghul — COMMENTED** [review]
Left a couple of comments, PTAL.

File: react/features/authentication/reducer.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable import/order */
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/authentication/reducer.ts:1]: This rule requires no empty lines between imports, but there's another rule that requires empty lines before comments (ts-ignore are comments)

File: react/features/base/conference/reducer.ts [hunk 2]
```diff
@@ -1,9 +1,13 @@
+/* eslint-disable import/order */
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/base/conference/reducer.ts:1]: Explained above

File: react/.eslintrc.js [hunk 3]
```diff
@@ -6,6 +6,29 @@ module.exports = {
...
+                '@typescript-eslint/ban-ts-comment': 'off',
+                '@typescript-eslint/no-empty-function': 'off',
+                '@typescript-eslint/ban-types': 'off',
+                '@typescript-eslint/no-use-before-define': 'off',
+                '@typescript-eslint/no-explicit-any': 'off',
+                'react/jsx-no-bind': 'off',
+                'no-prototype-builtins': 'off'
+            },
+            'plugins': [ '@typescript-eslint' ],
+            'extends': [
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/.eslintrc.js:23]: I don't know, but without these eslint doesn't understand interfaces

[the remark this answers is not in the document]
Comment (saghul) [reply react/.eslintrc.js:23]: Gotcha. We can look into it when moving the config to the eslint repo.

**Review by robertpin — APPROVED** [review]

**Pull request merged**